src/tests/ftest/util/server_utils.py

```diff
@@ -608,23 +608,25 @@ def get_single_system_state(self):
             raise ServerFailed("Multiple system states ({}) detected:\n  {}".format(states, data))
         return states.pop()
 
-    def check_rank_state(self, rank, valid_state, max_checks=1):
+    def check_rank_state(self, ranks, valid_states, max_checks=1):
         """Check the state of single rank in DAOS system.
 
         Args:
-            rankv(int): daos rank whose state need's to be checked
-            valid_state (str): expected state for the rank
+            rankv(list): daos rank list whose state need's to be checked
+            valid_states (list): list of expected states for the rank
             max_checks (int, optional): number of times to check the state
                 Defaults to 1.
         Raises:
             ServerFailed: if there was error obtaining the data for daos
                           system query
         Returns:
-            bool: returns True if there is a match for checked state,
-                  else False.
+            list: returns list of failed rank(s) if state does
+                  not match the expected state, otherwise returns empty list.
 
         """
         checks = 0
+        counter = 0
+        rank = None
```

Comment:
I don't think this needs to be defined upfront
```suggestion
```

Reply:
Done